File: frontend/src/services/authService.js
import apiClient from "../api/apiClient";

const authService = {
  async register(data) {
    const response = await apiClient.post("/auth/register", data);
    return response.data;
  },

  async login(data) {
    const response = await apiClient.post("/auth/login", data);
    return response.data;
  },
};

export default authService;

File: backend/src/main/java/com/ufinet/autos/service/interfaces/AuthService.java
package com.ufinet.autos.service.interfaces;

import com.ufinet.autos.dto.request.LoginRequest;
import com.ufinet.autos.dto.request.RegisterRequest;
import com.ufinet.autos.dto.response.AuthResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}
